scenarios/telemetry/dotnet/telemetry_producer/Worker.cs

```diff
@@ -22,28 +21,27 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         var cs = MqttConnectionSettings.CreateFromEnvVars(_configuration.GetValue<string>("envFile"));
         _logger.LogInformation("Connecting to {cs}", cs);
 
-        var mqttClient = new MqttFactory().CreateManagedMqttClient(MqttNetTraceLogger.CreateTraceLogger());
-
-        mqttClient.InternalClient.ConnectedAsync += async cea =>
+        var mqttClient = new MqttFactory().CreateMqttClient(MqttNetTraceLogger.CreateTraceLogger());
+        
+        mqttClient.DisconnectedAsync += async d =>
         {
-            _logger.LogInformation("Client {ClientId} connected: {ResultCode}", mqttClient.InternalClient.Options.ClientId, cea.ConnectResult.ResultCode);
+            await Task.Yield();
+            _logger.LogWarning("Disconnected from server with reason {reason} {ex}", d.Reason, d.Exception?.Message);
+        };
+        var connAck = await mqttClient.ConnectAsync(new MqttClientOptionsBuilder().WithConnectionSettings(cs).Build(), stoppingToken);
 
-            var telemetryPosition = new PositionTelemetryProducer(mqttClient.InternalClient);
+        _logger.LogInformation("Client {ClientId} connected: {ResultCode}", mqttClient.Options.ClientId, connAck.ResultCode);
 
-            while (!stoppingToken.IsCancellationRequested)
-            {
-                var pubAck = await telemetryPosition.SendTelemetryAsync(
-                    new Point(new Position(51.899523, -2.124156)), stoppingToken);
-                _logger.LogInformation("Message published with PUBACK {code}", pubAck.ReasonCode);
-                await Task.Delay(5000, stoppingToken);
-            }
-        };
+        var telemetryPosition = new PositionTelemetryProducer(mqttClient);
+
+        double counter = 1;
+        while (!stoppingToken.IsCancellationRequested)
+        {
+            var pubAck = await telemetryPosition.SendTelemetryAsync(
+                new Point(new Position(51.899523, -2.124156, counter++)), stoppingToken);
 
-        await mqttClient!.StartAsync(new ManagedMqttClientOptionsBuilder()
-            .WithClientOptions(new MqttClientOptionsBuilder()
-                .WithConnectionSettings(cs)
-                .Build())
-            .WithAutoReconnectDelay(TimeSpan.FromSeconds(5))
-            .Build());
+            _logger.LogInformation("Message '{mid}' published with PubAck '{code}'", pubAck.PacketIdentifier, pubAck.ReasonCode);
+            await Task.Delay(2000, stoppingToken);
```

Comment:
If `stoppingToken` is signaled, this will throw an `OperationCanceledException`. Do you want to catch that so it closes gracefully, the same if it happened to stop based on the `if` condition?

Reply:
This sample is just showing the happy path for the command, I'll add it once we have a complete design to deal with timeouts